src/executioners/OptimizeSolve.C

```diff
@@ -153,6 +153,10 @@ OptimizeSolve::taoSolve()
   ierr = TaoDestroy(&_tao);
   CHKERRQ(ierr);
 
+  ierr = MatDestroy(&_hessian);
+  LIBMESH_CHKERR(ierr);
+  _hessian = nullptr;
```

Comment:
Don't need this.

Reply:
The nullptr part?

Reply:
I need to delete the matrix to fix the memory leak in valgrind